cmd/vicadmin/server.go

```diff
@@ -75,6 +77,79 @@ const (
 	genericErrorMessage    = "Internal Server Error; see /var/log/vic/vicadmin.log for details" // for http errors that shouldn't be displayed in the browser to the user
 )
 
+// wrapper struct around net.Conn for our custom funcs
+type Conn struct {
+	net.Conn
+	b            byte
+	err          error
+	UncertainTLS bool
+}
+
+// Returns number of bytes read
+func (c *Conn) Read(b []byte) (int, error) {
+	// one time check to determine if TLS is in the connection
+	if c.UncertainTLS {
+		c.UncertainTLS = false
+		b[0] = c.b
```

Comment:
just for the sake of sanity, before this access we probably want to check the length of `b` and return an error if it's 0